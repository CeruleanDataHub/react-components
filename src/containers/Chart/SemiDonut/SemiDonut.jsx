import React from "react";

import { Chart } from "../Chart";

export const SemiDonut = ({ title, xAxis, series }) => {
  const options = {
    chart: {
      plotBackgroundColor: null,
      plotBorderWidth: 0,
      plotShadow: false
    },
    title: {
      text: title,
      align: "center",
      verticalAlign: "middle",
      y: 60
    },
    tooltip: {
      pointFormat: "{series.name}: <b>{point.percentage:.1f}%</b>"
    },
    accessibility: {
      point: {
        valueSuffix: "%"
      }
    },
    plotOptions: {
      pie: {
        dataLabels: {
          enabled: true,
          distance: -50,
          style: {
            fontWeight: "bold",
            color: "white"
          }
        },
        startAngle: -90,
        endAngle: 90,
        center: ["50%", "75%"],
        size: "110%"
      }
    },
    series: series.map(value => ({ ...value, type: "pie" }))
  };

  return (
    <Chart title={title} xAxis={xAxis} series={series} options={options} />
  );
};

SemiDonut.propTypes = {
  ...Chart.propTypes,
  /** Title of the chart */
  title: Chart.propTypes.title,
  /** Values for X axis */
  xAxis: Chart.propTypes.xAxis,
  /** Data to be graphed */
  series: Chart.propTypes.series
};

SemiDonut.defaultProps = {
  title: Chart.defaultProps.title,
  xAxis: Chart.defaultProps.xAxis,
  series: Chart.defaultProps.series
};
