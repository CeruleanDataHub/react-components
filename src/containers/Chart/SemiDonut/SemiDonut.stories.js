import React from "react";

import { SemiDonut } from "./SemiDonut";

export default {
  title: "SemiDonut",
  component: SemiDonut
};

const series = [
  {
    name: "Browser share",
    innerSize: "50%",
    data: [
      ["Chrome", 58.9],
      ["Firefox", 13.29],
      ["Internet Explorer", 13],
      ["Edge", 3.78],
      ["Safari", 3.42],
      {
        name: "Other",
        y: 7.61,
        dataLabels: {
          enabled: false
        }
      }
    ]
  }
];

export const SemiDonutStory = () => <SemiDonut title="Chart" series={series} />;
