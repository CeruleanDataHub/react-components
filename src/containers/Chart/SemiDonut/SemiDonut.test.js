import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { SemiDonut } from "./SemiDonut";

const series = [
  {
    name: "Browser share",
    innerSize: "50%",
    data: [
      ["Chrome", 58.9],
      ["Firefox", 13.29],
      ["Internet Explorer", 13],
      ["Edge", 3.78],
      ["Safari", 3.42],
      {
        name: "Other",
        y: 7.61,
        dataLabels: {
          enabled: false
        }
      }
    ]
  }
];

describe("Semi Donut", () => {
  it("should render chart", () => {
    const component = shallow(<SemiDonut />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render chart with data", () => {
    const component = shallow(<SemiDonut series={series} />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render chart with title and data", () => {
    const component = shallow(<SemiDonut title="Chart" series={series} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
