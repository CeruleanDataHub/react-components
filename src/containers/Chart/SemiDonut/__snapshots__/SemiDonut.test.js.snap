// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Semi Donut should render chart 1`] = `
<Chart
  containerProps={null}
  options={
    Object {
      "accessibility": Object {
        "point": Object {
          "valueSuffix": "%",
        },
      },
      "chart": Object {
        "plotBackgroundColor": null,
        "plotBorderWidth": 0,
        "plotShadow": false,
      },
      "plotOptions": Object {
        "pie": Object {
          "center": Array [
            "50%",
            "75%",
          ],
          "dataLabels": Object {
            "distance": -50,
            "enabled": true,
            "style": Object {
              "color": "white",
              "fontWeight": "bold",
            },
          },
          "endAngle": 90,
          "size": "110%",
          "startAngle": -90,
        },
      },
      "title": Object {
        "align": "center",
        "verticalAlign": "middle",
        "y": 60,
      },
      "tooltip": Object {
        "pointFormat": "{series.name}: <b>{point.percentage:.1f}%</b>",
      },
    }
  }
  series={Array []}
  title=""
  type="pie"
  xAxis={null}
/>
`;

exports[`Semi Donut should render chart with data 1`] = `
<Chart
  containerProps={null}
  options={
    Object {
      "accessibility": Object {
        "point": Object {
          "valueSuffix": "%",
        },
      },
      "chart": Object {
        "plotBackgroundColor": null,
        "plotBorderWidth": 0,
        "plotShadow": false,
      },
      "plotOptions": Object {
        "pie": Object {
          "center": Array [
            "50%",
            "75%",
          ],
          "dataLabels": Object {
            "distance": -50,
            "enabled": true,
            "style": Object {
              "color": "white",
              "fontWeight": "bold",
            },
          },
          "endAngle": 90,
          "size": "110%",
          "startAngle": -90,
        },
      },
      "title": Object {
        "align": "center",
        "verticalAlign": "middle",
        "y": 60,
      },
      "tooltip": Object {
        "pointFormat": "{series.name}: <b>{point.percentage:.1f}%</b>",
      },
    }
  }
  series={
    Array [
      Object {
        "data": Array [
          Array [
            "Chrome",
            58.9,
          ],
          Array [
            "Firefox",
            13.29,
          ],
          Array [
            "Internet Explorer",
            13,
          ],
          Array [
            "Edge",
            3.78,
          ],
          Array [
            "Safari",
            3.42,
          ],
          Object {
            "dataLabels": Object {
              "enabled": false,
            },
            "name": "Other",
            "y": 7.61,
          },
        ],
        "innerSize": "50%",
        "name": "Browser share",
      },
    ]
  }
  title=""
  type="pie"
  xAxis={null}
/>
`;

exports[`Semi Donut should render chart with title and data 1`] = `
<Chart
  containerProps={null}
  options={
    Object {
      "accessibility": Object {
        "point": Object {
          "valueSuffix": "%",
        },
      },
      "chart": Object {
        "plotBackgroundColor": null,
        "plotBorderWidth": 0,
        "plotShadow": false,
      },
      "plotOptions": Object {
        "pie": Object {
          "center": Array [
            "50%",
            "75%",
          ],
          "dataLabels": Object {
            "distance": -50,
            "enabled": true,
            "style": Object {
              "color": "white",
              "fontWeight": "bold",
            },
          },
          "endAngle": 90,
          "size": "110%",
          "startAngle": -90,
        },
      },
      "title": Object {
        "align": "center",
        "verticalAlign": "middle",
        "y": 60,
      },
      "tooltip": Object {
        "pointFormat": "{series.name}: <b>{point.percentage:.1f}%</b>",
      },
    }
  }
  series={
    Array [
      Object {
        "data": Array [
          Array [
            "Chrome",
            58.9,
          ],
          Array [
            "Firefox",
            13.29,
          ],
          Array [
            "Internet Explorer",
            13,
          ],
          Array [
            "Edge",
            3.78,
          ],
          Array [
            "Safari",
            3.42,
          ],
          Object {
            "dataLabels": Object {
              "enabled": false,
            },
            "name": "Other",
            "y": 7.61,
          },
        ],
        "innerSize": "50%",
        "name": "Browser share",
      },
    ]
  }
  title="Chart"
  type="pie"
  xAxis={null}
/>
`;
